src/piadm.sh

```diff
@@ -195,6 +196,106 @@ DEFAULT_URL_PREFIX=https://us-central.manta.mnx.io/Joyent_Dev/public/SmartOS/
 # Default path for piadm's configuration
 PIADM_CONF=/var/piadm/piadm.conf
 
+# fetch_csum
+#
+# Fetches checksums using a platform file as the key.
+#
+# Arguments:
+#   $1 - URL from where to fetch the PI from. The name of the PI
+#   	 is expected to be the last path component of the URL,
+#   	 or the literal string "latest".
+#
+# Environment Variables:
+#   PIADM_NO_SUM     - If set to 1, skips checksum validation.
+#   PIADM_SUM_URL    - If set, overrides the default URL used
+#                      the checksum file used for platform images.
+#
+# Notes:
+#   - If "latest" is used, up to 1024 PIs will be listed from ${URL_PREFIX}.
+#     The last one is assumed to be the latest.
+#   - On error, this function exits with code 1, following PIADM(8)
+#     convention: the error indicates a failure, but no changes were made
+#     to the system.
+declare csum_platform=""
+declare stamp=""
+fetch_csum() {
+	if [[ -z "$csum_platform" ]]; then
+		local platform_file
+		if [ "$1" != "latest" ]; then
+			IFS='/' read -ra array <<< "$1"
+			platform_file="${array[-1]}"
+			stamp="${array[-2]}"
+		else
+			#latest is a special case, we need find out stamp
+			stamps_url="${URL_PREFIX%/}?limit=1024"
+			stamp=$("${CURL[@]}" "${stamps_url}" |\
+				json -ga -c "this.name.match(/Z$/)" | json -ag name |\
+				sort  | tail -1; exit "${PIPESTATUS[0]}")
+			code=$?
+			if [[ $code -ne 0 ]]; then
+				eecho "Curl failed fetching PI data from ${stamps_url}"
+				return 1
+			fi
+			platform_file="smartos-${stamp}.iso"
+		fi
+		if [[ -z ${PIADM_SUM_URL} ]];then
+			local csum_url="${URL_PREFIX}${stamp}/md5sums.txt"
+		else
+			local csum_url="${PIADM_SUM_URL}"
+		fi
+		csum_platform=$("${CURL[@]}"  "${csum_url}" |\
+			awk -v pattern="${platform_file}"\
+			'$0 ~ pattern { print $1; exit}'; exit "${PIPESTATUS[0]}")
+		code=$?
+		if [[ $code -ne 0 ]]; then
+			eecho "fetching checksums from ${csum_url}"
+			#force re-calculation of checksum on the next call.
+			csum_platform=""
+			return 1
+		fi
+		echo "${csum_platform}"
+		return 0
+	else
+		echo "${csum_platform}"
+		return 0
+	fi
+}
+
+# validate_csum
+#
+# $1 is the URL from where the PI was downloaded.
+# $2 is the on disk PI image that was download from $1.
+#
+# Error code will always be 1, following the PIADM(8) convention where
+# a return code of 1 means: an error has occurred, but no change was made.
+# Environment variable PIADM_DIGEST_ALGORITHM controls the checksum
+# algorithm used by DIGEST(1), by default md5 is used.  
```

Comment:
You fixed it with `piadm`(8) in the man page, but digest(1) needs to be lowercase.